cloudinary-core/src/main/java/com/cloudinary/Uploader.java

```diff
@@ -516,4 +516,23 @@ public String imageUploadTag(String field, Map options, Map<String, Object> html
     public Map deleteByToken(String token) throws Exception {
         return callApi("delete_by_token", ObjectUtils.asMap("token", token), ObjectUtils.emptyMap(), null);
     }
+
+    /**
+     * Populates metadata fields with the given values. Existing values will be overwritten.
+     * @param metadata a map of field name and value.
+     * @param publicIds the public IDs of the resources to update
+     * @param options additional options passed to the request
+     * @return a list of public IDs that were updated
+     * @throws IOException
+     */
+    public Map updateMetadata(Map metadata, String[] publicIds, Map options) throws IOException {
```

Comment:
@nitzanj, don't we want to have a single resource update as well? 
Also, maybe we should have api.updateField(publicId, fieldId, value, options)

Reply:
This mirrors the actual API. We can add helper methods if you think it's needed but that will push the release date, and we'll need to get/write a spec for those before moving on.

Reply:
Let's leave it for now and add those when users will ask.